src/base_model.jl

```diff
@@ -11,6 +11,12 @@ Creates the base optimization model for all the EC models
 data: structure of data
 '''
 """
+#This function allow to get the profile of each users related to their market type (e.g. commercial, non_commercial)
+function market_profile_by_user(ECModel, u_name, profile_name)

```

Comment:
perfect! Minor comments:
- you added this function after the docstring of build_base_model!, please, move this function well below the definition
- Please, add `ECModel::AbstractEC`
- the use of `ECModel.market_data[user_market_type]` may lead to a difficult error to read if the type of market is not provided.
  you may create an auxiliary variable `market_data_type = field(market_data, user_market_type, desc="Missing market type definition '$user_market_type'")` to handle that exeption
- Please add a docstring  to this function similarly to the previous ones.

Reply:
Updated in last changes

Reply:
Mmmm maybe it is not updated, but I don't see the changes